### PR TITLE
#693 Renaming test entities

### PR DESCRIPTION
**Pull request by gunnarmorling**

### CODE REVIEW

File: parent/pom.xml [hunk 1]
```diff
@@ -46,7 +46,7 @@
         <org.apache.maven.plugins.surefire.version>2.18.1</org.apache.maven.plugins.surefire.version>
         <org.apache.maven.plugins.javadoc.version>2.10.3</org.apache.maven.plugins.javadoc.version>
         <org.springframework.version>4.0.3.RELEASE</org.springframework.version>
-        <org.eclipse.tyco.compiler-jdt.version>0.23.1</org.eclipse.tyco.compiler-jdt.version>
+        <org.eclipse.tycho.compiler-jdt.version>0.23.1</org.eclipse.tycho.compiler-jdt.version>
```
Comment (agudian) [inline parent/pom.xml]: hehe, right! Thanks! :+1:

**Comment by agudian** [comment]
Great, now _I'm_ confused and don't understand the example anymore... :wink:

Perhaps we need to do what Sjaak suggested and draw up a good-ol' diagram...? Any idea on where to place that? Just to the wiki maybe?

**Comment by gunnarmorling** [comment]
I recommend to check out the type hierarchy in Eclipse ;) But yes, a diagram would be nice.

File: processor/src/test/java/org/mapstruct/ap/test/abstractclass/generics/AnimalKey.java [hunk 2]
```diff
@@ -22,14 +22,14 @@
  * @author Andreas Gudian
  *
  */
-public class ItemC extends ItemB {
+public class AnimalKey extends KeyOfAllBeings {
     private boolean typeParameterIsResolvedToItemC;
 
-    public boolean typeParameterIsResolvedToItemC() {
+    public boolean typeParameterIsResolvedToAnimalKey() {
         return typeParameterIsResolvedToItemC;
```
Comment (agudian) [inline processor/src/test/java/org/mapstruct/ap/test/abstractclass/generics/AnimalKey.java]: Could you also rename the field to `typeParameterIsResolvedToAnimalKey`?

File: processor/src/test/java/org/mapstruct/ap/test/abstractclass/generics/Child.java [hunk 3]
```diff
@@ -22,14 +22,14 @@
...
  *
  */
-public class SourceWithItemB extends AbstractClassExposingItemB {
-    private ItemB item;
+public class Child extends AbstractHuman {
+    private KeyOfAllBeings item;
 
-    public ItemB getItem() {
+    public KeyOfAllBeings getKey() {
         return item;
```
Comment (agudian) [inline processor/src/test/java/org/mapstruct/ap/test/abstractclass/generics/Child.java]: Rename field `item` to `key`?

File: processor/src/test/java/org/mapstruct/ap/test/abstractclass/generics/AnimalKey.java [hunk 4]
```diff
@@ -22,14 +22,14 @@
...
     }
 
-    public void setTypeParameterIsResolvedToItemC(boolean typeParameterIsResolvedToItemC) {
+    public void setTypeParameterIsResolvedToAnimalKey(boolean typeParameterIsResolvedToItemC) {
         this.typeParameterIsResolvedToItemC = typeParameterIsResolvedToItemC;
```
Comment (agudian) [inline processor/src/test/java/org/mapstruct/ap/test/abstractclass/generics/AnimalKey.java]: The parameter could also be renamed, for the sake of consistency...

File: processor/src/test/java/org/mapstruct/ap/test/abstractclass/generics/Elephant.java [hunk 5]
```diff
@@ -22,16 +22,16 @@
...
  */
-public class SourceWithItemC extends AbstractClassExposingItemC {
-    private ItemC item;
+public class Elephant extends AbstractAnimal {
+    private AnimalKey item;
 
     @Override
-    public ItemC getItem() {
+    public AnimalKey getKey() {
         return item;
```
Comment (agudian) [inline processor/src/test/java/org/mapstruct/ap/test/abstractclass/generics/Elephant.java]: `s/item/key/g` again... :)

File: processor/src/test/java/org/mapstruct/ap/test/abstractclass/generics/KeyOfAllBeings.java [hunk 6]
```diff
@@ -22,14 +22,14 @@
  * @author Andreas Gudian
  *
  */
-public class ItemB extends ItemA {
+public class KeyOfAllBeings extends Key {
     private boolean typeParameterIsResolvedToItemB;
 
-    public boolean typeParameterIsResolvedToItemB() {
+    public boolean typeParameterIsResolvedToKeyOfAllBeings() {
         return typeParameterIsResolvedToItemB;
```
Comment (agudian) [inline processor/src/test/java/org/mapstruct/ap/test/abstractclass/generics/KeyOfAllBeings.java]: `s/typeParameterIsResolvedToItemB/typeParameterIsResolvedToKeyOfAllBeings/g`

File: processor/src/test/java/org/mapstruct/ap/test/abstractclass/generics/Target.java [hunk 7]
```diff
@@ -23,22 +23,22 @@
  *
  */
 public class Target {
-    private ItemC itemC;
-    private ItemB itemB;
+    private AnimalKey itemC;
+    private KeyOfAllBeings itemB;
```
Comment (agudian) [inline processor/src/test/java/org/mapstruct/ap/test/abstractclass/generics/Target.java]: The properties (including getter/setter) should be named `animalKey` and `keyOfAllBeings` to make it easier to grasp in the mapper definition.

**Comment by agudian** [comment]
I've added some more suggestions to make the naming consistent. Other than that, I like it... :+1:

**Comment by gunnarmorling** [comment]
Thanks for checking. Yes, quite some left-overs indeed. I'll fix it and
merge after 1.0 Final.

**Comment by agudian** [comment]
Dear Jenkins, retest this, please.

Let's see how long it'll take on the second take.

**Comment by sjaakd** [comment]
cool :+1:

**Comment by gunnarmorling** [comment]
Rebased and applied.

**Pull request closed without merging**